### PR TITLE
fix(FP): Several FPs not suitable for our automation

### LINKED ISSUE #4993: [FP]: False Positive for cross-platform-appiancorp-json-parser-java.jar
### Package URl

pkg:maven/com.appiancorp/appiancorp-json-parser-java@23.1.0.135

### CPE

cpe:2.3:a:json_project:json:23.1.0.135:*:*:*:*:*:*:*  

### CVE

CVE-2020-10675 

### ODC Integration

{"label"=>"CLI"}

### ODC Version

7.0.4

### Description

we are using our own custom implementation of JSON parser but tool is giving CVE for this cross-platform-appiancorp-json-parser-java.jar 
Here is the reference : https://github.com/buger/jsonparser/issues/188

### LINKED ISSUE #5381: [FP]: apollographql flagged with cpe:2.3:a:apollo_project:apollo:2.5.6:*:*:*:*:*:*:*
### Package URl

pkg:maven/com.apollographql.apollo/apollo-api-jvm@2.5.6

### CPE

cpe:2.3:a:apollo_project:apollo:2.5.6:*:*:*:*:*:*:*

### CVE

CVE-2015-10043 

### ODC Integration

{"label"=>"Gradle Plugin"}

### ODC Version

8.0.1

### Description

apollographql flagged with cpe:2.3:a:apollo_project:apollo:2.5.6:*:*:*:*:*:*:*

Also being flagged similarly:

- pkg:maven/com.apollographql.apollo/apollo-http-cache-api@2.5.6
- pkg:maven/com.apollographql.apollo/apollo-normalized-cache-api-jvm@2.5.6
- pkg:maven/com.apollographql.apollo/apollo-normalized-cache-jvm@2.5.6
- pkg:maven/com.apollographql.apollo/apollo-runtime@2.5.6

### LINKED ISSUE #5454: [FP]:  vy/flatbuffers(https://github.com/vy/flatbuffers) is not hughsk flat(https://github.com/hughsk/flat)
### Package URl

pkg:maven/com.vlkan/flatbuffers@1.2.0-3f79e055

### CPE

cpe:2.3:a:flat_project:flat:1.2.0.3.f79.e055:*:*:*:*:*:*:*

### CVE

CVE-2020-36632

### ODC Integration

{"label"=>"Gradle Plugin"}

### ODC Version

8.1.0

### Description

It seems that vy/flatbuffers is being confused with hughsk flat according to the cve description:
[CVE-2020-36632]
A vulnerability, which was classified as critical, was found in hughsk flat up to 5.0.0. This affects the function unflatten of the file index.js. The manipulation leads to improperly controlled modification of object prototype attributes ('prototype pollution'). It is possible to initiate the attack remotely. Upgrading to version 5.0.1 is able to address this issue. The name of the patch is 20ef0ef55dfa028caddaedbcb33efbdb04d18e13. It is recommended to upgrade the affected component. The identifier VDB-216777 was assigned to this vulnerability.

### LINKED ISSUE #5485: [FP]: jaxb-core detected as Java_SE
### Package URl

pkg:maven/com.sun.xml.bind/jaxb-core@2.3.0

### CPE

cpe:2.3:a:oracle:java_se:2.3.0:*:*:*:*:*:*:*

### CVE

_No response_

### ODC Integration

{"label"=>"CLI"}

### ODC Version

8.1.0

### Description

Hello,

The 2 folllowing Maven packages:
- pkg:maven/com.sun.xml.bind/jaxb-core@2.3.0
- pkg:maven/org.glassfish.jaxb/jaxb-core@2.3.0

are incorrectly associated to CPE `cpe:2.3:a:oracle:java_se:2.3.0:*:*:*:*:*:*:*`

probably because of the following hint evidence:

image
image

Thank you.

Regards,

### LINKED ISSUE #5489: [FP]: Apache Calcite Avatica
### Package URl

pkg:maven/org.apache.calcite.avatica/avatica-core@1.23.0

### CPE

`cpe:2.3:a:apache:calcite:1.23.0:*:*:*:*:*:*:*`

### CVE

CVE-2022-39135

### ODC Integration

{"label"=>"Maven Plugin"}

### ODC Version

8.1.0

### Description

Apache Calcite Avatica is a subproject of Apache Calcite with its own cpe: `cpe:2.3:a:apache:apache_calcite_avatica:1.23.0:*:*:*:*:*:*:*`

### LINKED ISSUE #5502: [FP]: json-java CVE flagged for multiple unrelated Java JSON projects
### Package URl

- Jackson: pkg:maven/com.fasterxml.jackson.core/jackson-core@2.14.0-rc1
- Jakarta JSON: pkg:maven/org.glassfish/jakarta.json@2.0.1
- Javax JSON: pkg:maven/org.glassfish/javax.json@1.0.4
- JsonLD-Java: pkg:maven/com.github.jsonld-java/jsonld-java@0.13.4

### CPE

- Jackson:
  - `cpe:2.3:a:fasterxml:jackson-modules-java8:2.14.0:rc1:*:*:*:*:*:*`
  - `cpe:2.3:a:json-java_project:json-java:2.14.0:rc1:*:*:*:*:*:*`
- Jakarta JSON:
  - `cpe:2.3:a:eclipse:glassfish:2.0.1:*:*:*:*:*:*:*`
  - `cpe:2.3:a:json-java_project:json-java:2.0.1:*:*:*:*:*:*:*`
- Javax JSON:
  - `cpe:2.3:a:json-java_project:json-java:1.0.4:*:*:*:*:*:*:*`
- JsonLD-Java: 
  - `cpe:2.3:a:json-java_project:json-java:0.13.4:*:*:*:*:*:*:*`

### CVE

CVE-2022-45688

### ODC Integration

{"label"=>"CLI"}

### ODC Version

7.2.1

### Description

Neither jsonld-java nor hutools (the originally affected project) appear in the full dependency tree of the Clojure/Java project I'm working on. Inspecting the dependencies, neither Jackson nor jsonld-java use java-json or hutools as dependencies. Not sure if jakarta/javax.json use java-json but if the other two libraries have false positives it's probable that their CVEs are FPs as well.

### PR DESCRIPTION
**Pull request by aikebah**
## Description of Change

FP suppressions for which either the automation could not run, or the automation yields an inappropriate suppression

## Have test cases been added to cover the new functionality?

no

### CODE REVIEW

**Comment by aikebah** [comment]
@jeremylong given the amount of libraries hit by #5502 I think it would be good to get an 8.1.1 release out when this PR is integrated

**Comment by adam-siklosi** [comment]
Hi @aikebah

When I saw the release notes for #4723 I thought the age of updating DependencyCheck version for false positive suppressions is over.
Was that a misunderstanding from my side or you are still working on some aspect of this feature?

Thanks!
Adam

**Comment by aikebah** [comment]
@adam-siklosi Not all suppressions are feasible for the automation. The automation is allowing *most* suppressions (the simple ones - single library, improper CPE - which is the majority of the FP reports) to have a much faster roundtrip time without a need to release (simple suppressions for one library can be handled by automation to be pushed to the published suppressions).
The more complex ones are still only processed in regular development. There is no automation in place to process manual additions to published suppressions.

File: core/src/main/resources/dependencycheck-base-suppression.xml [hunk 1]
```diff
@@ -6180,4 +6180,211 @@
...
+        <cpe>cpe:/a:freebsd:freebsd</cpe>
+    </suppress>
+    <suppress base="true">
+        <notes><![CDATA[
+   FP per issue #5506
+   ]]></notes>
+        <packageUrl regex="true">^pkg:maven/io\.kamon/kamon-prometheus_2\.13@.*$</packageUrl>
+        <cpe>cpe:/a:prometheus:prometheus</cpe>
+    </suppress>
+    <!-- end generated suppressions added to main in 8.1.1 -->
```
Comment (jeremylong) [inline core/src/main/resources/dependencycheck-base-suppression.xml]: Copied one too many lines.
```suggestion
```

**Review by jeremylong — APPROVED** [review]

**Comment by jeremylong** [comment]
@aikebah I'll release the new version shortly.

**Comment by Marcono1234** [comment]
@aikebah, in cases where automation cannot create the suppression, would it not be possible to create the pull request manually against the `generatedSuppressions` branch (or manually edit the bot generated suppression as it was done in https://github.com/jeremylong/DependencyCheck/issues/5122#issuecomment-1340174717)?

**Pull request merged**